### ISSUE TITLE
Duplicate "Duplicate identifier" errors between method and getter

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#51437]
[Playground Link](https://www.typescriptlang.org/play?#code/MYGwhgzhAEAa0G8CwAoa7oHpPQIzQFMAnIgeyNQ2gDNTSAKASkUqoyIIBcBXIgOzwBuVugC+qEVhwAmQiXIRJAcy406TFmjboOPfkMniUooA)

Notice that the `get` accessor below gets two error spans.

```ts
class X {
    foo() {
//  ~~~ Duplicate identifier 'foo'.
        return 1;
    }

    get foo() {
    //  ~~~ Duplicate identifier 'foo'.
    //  ~~~ Duplicate identifier 'foo'.
        return 1;
    }
}
```

Expected:

```ts
class X {
    foo() {
//  ~~~ Duplicate identifier 'foo'.
        return 1;
    }

    get foo() {
    //  ~~~ Duplicate identifier 'foo'.
        return 1;
    }
}
```

**Comment by jwertherUM** [comment]
Hi, I'm fairly new to open source but would like to take this issue. Is it still up for grabs?

**Comment by abuob** [comment]
I tried my hand a bit on this issue, here are the outcomes (first time working on this repository, please be blunt with me in case something is amiss): 

It seems that both the binder and the checker report the same diagnostics for the second declaration, at least that was what I got when I set a breakpoint here when debugging the language service: [program.ts#L2404-L2405](https://github.com/microsoft/TypeScript/blob/main/src/compiler/program.ts#L2404-L2405)

Therefore, a possible fix might be to simply remove the duplicates - I guess filtering by message-type and start-property might suffice? From a naming-point-of-view, it might make sense to do this in `getMergedBindAndCheckDiagnostics` (see [program.ts#L2415](https://github.com/microsoft/TypeScript/blob/main/src/compiler/program.ts#L2415)), but as I haven't seen similar logic there, it might be the wrong place/approach altogether? It would however lead to the outcome described as expected per the above. 

Another option would be to treat the case similar to other scenarios like duplicate method declarations or property declarations; see e.g. the playground-examples here: [Playground link with similar examples](https://www.typescriptlang.org/play?#code/MYGwhgzhAEAa0G8CwAoa7oHpPQIzQFMAnIgeyNQ2gDNTSAKASkUqoyIIBcBXIgOzwBuVugC+qEVhwAmQiXIRJAcy406TFmjboOPfkMniUk7NAAqACwJqQIUgHcAlnyXRgkAjDAdoEUgFsCewswTl9Hf0dwIgAaaAAjbjCAE1JoPlIwkIA3a04raGTuAAcQR3dOR1I+AFpisniQAn8ALgktdFMAERKyiutqbj5gSuqa5IJQb1CqvkFoEJhHThgHAWIyIhrAqDAVSXjvDWQOtl1eAVxhU6MDo+YEaEkqUwA5TIIW6GdqDYJk6Dnfg1TgAT2K1kcMAARHxuP54sRodB6HZSABrEq+Ag4-wwThpIECfLWQL5UgAsDxUi5RjPdhcC7QWGZGpgGpwhFIwztF44AAKZAhRFB40m0Rm1Tap0OAC9oABePB3eVK2SmADyfBAoOgJOxwGqAImUyIkoEi2gYHW8goxhQoiAA)

These cases don't seem to be considered an issue at all from the binders point of view; only the checker reports errors. However, presumably because of the overwrite in the symbol-table, the inferred types etc. are reported wrong on the duplicate.
If only reporting the second declaration as an error and having "wrong lookups" for it is fine, it would be an option to exclude the diagnostics done by the binder altogether. I tried changing the following lines as follows:

[types.ts#L5406-L5407](https://github.com/microsoft/TypeScript/blob/main/src/compiler/types.ts#L5406-L5407)
```typescript
    GetAccessorExcludes = Value & ~SetAccessor & ~Method,
    SetAccessorExcludes = Value & ~GetAccessor & ~Method,
```
This does break a few tests, mostly in ways as expected per the above though. Not sure if this is in line with your design philosophy regarding behavior in error-cases?

**TL;DR:**
Issue seems to be that both the binder and checker report the same diagnostics. Some approaches:
1. Filter merged outcome for duplicate entries
2. Limit error-reporting for this case to the checker (a bit more in line with e.g. duplicate property declaration)
3. Create custom message/handling of this case
4. (other things I probably missed)

As I'm already down the rabbit hole, I'd love to submit a PR for this, I'm not quite sure how to proceed here though, there's probably plenty of stuff that I missed. Feedback very much welcome, many thanks in advance for a reply! :)

**Comment by gmanninglive** [comment]
Just submitted a PR for this #51594
Super simple fix, had one fourslash test fail as a result but I believe it is behaving as expected with change of index?

And thanks @abuob for digging into the issue and finding the source!

**Comment by abuob** [comment]
@gmanninglive Ah, there's already a `sortAndDeduplicateDiagnostics`-method, I missed this one, hence I wasn't sure if this is the right approach :P Nice!
